app/services/authn_context_resolver.rb

```diff
@@ -54,18 +66,51 @@ def no_identity_proofing_vot
     end
   end
 
+  # Returns a copy of acr_values where the appropriate IAL authentication context class reference
+  # name replaces <tt>Saml::Idp::Constants::IAL2_BIO_PREFERRED_AUTHN_CONTEXT_CLASSREF</tt>.
+  # @note This implementation will be replaced with a more robust long-term
+  # solution.
+  # @param [String] acr_values
+  def initialize_acr_values(acr_values)
+    if sp_in_biometric_pilot? && acr_values.present?
+      biometric_proofing_requested = acr_values.include?(Saml::Idp::Constants::IAL2_BIO_PREFERRED_AUTHN_CONTEXT_CLASSREF)
+      if biometric_proofing_requested && user&.identity_verified_with_biometric_comparison?
```

Comment:
@jmhooper just wanted to check if we'd have this level of user insight at this point in the code

Reply:
oh nevermind, it looksl ike this is similar to `selected_vtr_parser_result_from_vtr_list`, sorry!

Reply:
question -- could we have made the two different components (similar to the VoT implementation) and just utilized that same method to parse the components?

Reply:
It seems odd to me to replace the ACR values that are requested as a side-effect of initialization.

Did we consider appluinh this logic in a method similar to `decorate_acr_result_with_sp_defaults`? I'd imagine something like `decorate_acr_result_with_user_context` or something like that? That way what we return from `acr_values` matches what was passed in as the `acr_values` arg and carries through to the `component_values` on the result.

Reply:
so i think the string manipulation is a consequence of `String specifying an Authentication Context Class Reference value that identifies the Authentication Context Class that the authentication performed satisfied.` 

ie, the acr_value is not the value passed in, but the value satisfied. `preferred` doesn't really mean anything in that context -- if the user was satisfied via biometric comparison vs our "base" idv is more meaningful and communicates more to the partner. does that make sense?

Reply:
i am going to try out this decorator class @lmgeorge

Reply:
we had talked yesterday about string manipulation not feeling great -- lmk what you think about the changes, i think the decorator feels a little better and is more in line with how this class was intended to work!

Reply:
@Sgtpluck - I like using the decorator, but it introduces an issue that I flagged in a different [comment](https://github.com/18F/identity-idp/pull/10948#discussion_r1681870103) and goes back to our conversation about how the vector-style component values work subtly different than the ACR component values. 

Since each component for the ACR values has no dependencies (i.e., `implied_component_values`) with a flat list of requirements and the components are immutable, we either have to hoist the business logic of "coupled" requirements (e.g., `:biometric_comparison` and `:two_pieces_of_fair_evidence`) to `AuthnContextResolver` or have the requirements be evaluated dynamically on `ComponentValue` and `Parser::Result`.

I think hoisting that business logic is fine for now, but we'll need to rework the design later.